### PR TITLE
Used useReducer instead of useState.

### PR DESCRIPTION
**Pull request by syed-shaida-hussain**
Removed useState and used useReducer instead.

### CODE REVIEW

**Review by nikhil-sharma-nks — APPROVED** [review]
Looks good to merge. After using reducer your code is more structured and easy to predict and understand. Good work👍

File: src/components/Header.js [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import { Link } from "react-router-dom";
-import {useCart} from "../contexts/cart-context/cart-context"
-import {useWishlist} from "../contexts/wishlist-context/wishlist-context"
+import { useServices } from "../contexts/productContext/productContext";
 
 const Header = () => {
-    const {cartCount} = useCart()
-    const {wishCount} = useWishlist()
+    const { serviceState : { cartItems , wishItems } , dispatchService } = useServices()
```
Comment (shrushti2000) [inline src/components/Header.js:5]: nice that you have destructured object

File: src/contexts/productContext/productContext.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { createContext , useContext , useReducer } from "react"
+import { serviceReducer } from "../../reducers/service-reducer/service-reducer"
+
+const ServiceContext = createContext()
+
+
+
```
Comment (shrushti2000) [inline src/contexts/productContext/productContext.js]: can remove this unused lines

**Review by shrushti2000 — APPROVED** [review]
Good that you have used useReducer instead of useState. Good code optimization. Can be merged

**Pull request merged**